community/io/src/main/java/org/neo4j/io/fs/FileUtils.java

```diff
@@ -628,4 +630,77 @@ public MaybeWindowsMemoryMappedFileReleaseProblem( IOException e )
             super( e );
         }
     }
+
+    /**
+     * Calculates the size of a given directory or file.
+     *
+     * @param path to the file or directory.
+     * @return the size, in bytes, of the file or the total size of the content in the directory, including
+     * subdirectories.
+     */
+    public static long size( File path )
+    {
+        final MutableLong size = new MutableLong();
+
+        try
+        {
+            Files.walkFileTree( path.toPath(), new SimpleFileVisitor<Path>()
+            {
+                @Override
+                public FileVisitResult visitFile( Path file, BasicFileAttributes attrs )
+                {
+                    size.add( attrs.size() );
+                    return FileVisitResult.CONTINUE;
+                }
+
+                @Override
+                public FileVisitResult visitFileFailed( Path file, IOException exc )
+                {
+                    return FileVisitResult.CONTINUE; // Ignore exception
+                }
+
+                @Override
+                public FileVisitResult postVisitDirectory( Path dir, IOException exc )
+                {
+                    return FileVisitResult.CONTINUE; // Ignore exception
+                }
+            } );
+        }
+        catch ( IOException e )
+        {
+            throw new AssertionError( "Exceptions during file size calculations should never happen", e );
```

Comment:
`walkFileTree` is guaranteed to not throw IOException unless the visitor does. This will be removed anyway since the method is not used anymore.